drivers/pse/si3474/si3474.c

```diff
@@ -0,0 +1,383 @@
+	int mask = 0x1;
+
+	res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, INTERRUPT_REGISTER,
+				  &pse_interrupt_buffer);
+	if (res != 0) {
+		LOG_WRN("Si3474 reading interrupt status failed");
+	}
+
+	for (int cnt = 0; cnt <= 8; cnt++) {
+		switch (pse_interrupt_buffer & mask) {
+		case POWER_ENABLE_CHANGE_IT:
+			mask <<= 1;
+			cnt++;
+		case POWER_GOOD_CHANGE_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, POWER_EVENT_REG,
+						  &events[POWER_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+		case DISCONNECT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address,
+						  DISCONNECT_PCUT_FAULT_REG,
+						  &events[DISCONNECT_PCUT_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, SUPPLY_EVENT_REG,
+						  &events[SUPPLY_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case DETECT_CC_DONE_IT:
+			mask <<= 1;
+			cnt++;
+		case CLASS_DONE_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, CLASS_DETECT_EVENT_REG,
+						  &events[CLASS_DETECT_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case P_I_FAULT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address,
+						  DISCONNECT_PCUT_FAULT_REG,
+						  &events[DISCONNECT_PCUT_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, ILIM_START_FAULT_REG,
+						  &events[ILIM_START_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, SUPPLY_EVENT_REG,
+						  &events[SUPPLY_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case START_EVENT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, ILIM_START_FAULT_REG,
+						  &events[ILIM_START_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, POWER_ON_FAULT_REG,
+						  &events[POWER_ON_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case SUPPLY_EVENT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, SUPPLY_EVENT_REG,
+						  &events[SUPPLY_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+		}
+		mask <<= 1;
+	}
+	return res;
+}
+
+static int si3474_set_events(const struct device *dev, uint8_t events)
+{
+	const struct si3474_config *cfg = dev->config;
+	uint16_t dev_address = cfg->i2c.addr;
+	int res;
+
+	res = i2c_reg_write_byte(cfg->i2c.bus, dev_address, INTERRUPT_MASK, events);
+	if (res != 0) {
+		LOG_WRN("Si3474 setting events failed!");
+		return res;
```

Comment:
Can be removed

Reply:
Maybe I should rename it to activate events? si3474_set_events() is configuring which types/events will generates the interrupt. Generated events are fetched from the sensor by si3474_get_events();

Reply:
I just meant that the `return` could be removed because there's a return right after.  I think the function name is fine.